docs/functionaldatatypes/validation.md

```diff
@@ -0,0 +1,286 @@
+
+## Combining Validation Values
+
+We can also combine `Validation` values.
+
+### Accumulating Errors
+
+The most common way to do this is with the `validateWith` operator we saw above.
+
+```scala mdoc
+object Validation {
+  def validateWith[E, A, B, C](left: Validation[E, A], right: Validation[E, B])(f: (A, B) => C): Validation[E, C] =
+    ???
+}
+```
+
+This will check each of the original `Validation` values and if they are both successes combine their values with the function `f` to produce a new successful `Validation` value. If one or more of the original `Validation` values are failures then the result will be a failure with all of the failures of both original `Validation` values.
+
+There are variants of `validateWith` for combining up to twenty two different `Validation` values so even if you are working with large cases classes with many fields you can still use `validateWith` to combine them.
+
+There is also a `validate` variant that combines multiple `Validation` values but does not take a combining function. In this case the returned `Validation` just contains a tuple with all the original values and you can combine them yourself later using `map`.
+
+If you have a whole collection of values you want to validate you can use the `collectAllPar` operator to validate them all at the same time.
+
+### Chaining Validations
+
+The second way we might want to combine `Validation` values is by returning another `Validation` value based on the result of the first validation value. For example, we might first try to parse a string into an integer and then validate that the integer is within a reasonable range.
+
+```scala mdoc:reset
+import zio.prelude.Validation
+
+def validateInt(s: String): Validation[String, Int] =
+  Validation(s.toInt).mapError(_.getMessage)
+
+def validateAge(age: Int): Validation[String, Int] =
+  Validation.fromPredicateWith(s"Age $age was less than zero")(age)(_ >= 0)
+
+def validateAgeString(s: String): Validation[String, Int] =
+  for {
+    n <- validateInt(s)
+    _ <- validateAge(n)
+  } yield n
+```
+
+Here the second validation depended on the first one because if we could not parse the string into an integer then we would not even have an integer to validate. Just like with other data types we can express this with the `flatMap` operator and use a _for comprehension_ to chain multiple validations together.
+
+Note that when we chain validations like this we only do the second validation if the first one is successful so we will never see both errors here. If all we are doing is chaining then we don't actually need `Validation` and could just use `Either`.
+
+However, frequently we want to do some validations "in parallel" and other "sequentially" and having `flatMap` on `Validation` makes it very easy for us to do this. Note that when we say in parallel here no actual parallelism is going on here, we just mean that we are performing both validations even if one of them fails.
```

Comment:
pedantic (API related, not doc related): that parallel wording feels unfortunate to me, but I hesitate to suggest it warrants a binary compatibility break in the future.

Reply:
Changed to accumulating errors and chaining validations.